package/src/dev/hmr-integration.ts

```diff
@@ -0,0 +1,17 @@
+import { z } from "astro/zod";
+import { defineIntegration } from "../core/define-integration.js";
+import { watchDirectory } from "../utilities/watch-directory.js";
+
+export const hmrIntegration = defineIntegration({
+	name: "aik/hmr",
+	optionsSchema: z.object({
+		directory: z.string(),
+	}),
+	setup({ options }) {
+		return {
+			"astro:config:setup": (params) => {
+				watchDirectory(params, options.directory);
```

Comment:
If `hmrIntegration` just calls `watchDirectory` it still has the problem of not detecting any files _added_ to the directory.

Might be worth documenting it as a known limitation. Workarounds for it don't really seem worth it and a proper solution requires plenty of changes in core (maybe we should do that).

Reply:
You're right. I think we should definitely look into the glob proposal soon. Otherwise yeah, documenting seems enough. Do you know if there's a way through the vite dev server (`astro:server:setup` or `astro:server:start`) to make astro restart? because we could use vite file watcher if we wanted. But yeah seems like a lot of work for what it's worth

Reply:
Docs updated, let me know!

Reply:
It might be possible, but I haven't explored that part of the API too deeply yet so I don't know.